### ISSUE TITLE
[BUG] 微调lora 报错<title>

### ISSUE THREAD

**Issue by mingo-doer** [OpenBMB/MiniCPM-V#346]
### 是否已有关于该错误的issue或讨论？ | Is there an existing issue / discussion for this?

- [X] 我已经搜索过已有的issues和讨论 | I have searched the existing issues / discussions

### 该问题是否在FAQ中有解答？ | Is there an existing answer for this in FAQ?

- [X] 我已经搜索过FAQ | I have searched FAQ

### 当前行为 | Current Behavior

tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
{'loss': 2.4388, 'grad_norm': 0.0, 'learning_rate': 0, 'epoch': 0.04}                                                                                                 
  0%|                                                                                                                             | 6/10000 [00:20<8:58:00,  3.23s/it]tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
{'loss': 2.3092, 'grad_norm': 0.0, 'learning_rate': 0, 'epoch': 0.05}                                                                                                 
  0%|                                                                                                                             | 7/10000 [00:23<8:47:42,  3.17s/it]tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
{'loss': 2.4331, 'grad_norm': 0.0, 'learning_rate': 0, 'epoch': 0.05}                                                                                                 
  0%|                                                                                                                             | 8/10000 [00:26<8:30:50,  3.07s/it]tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
{'loss': 2.3796, 'grad_norm': 0.0, 'learning_rate': 0, 'epoch': 0.06}                                                                                                 
  0%|                                                                                                                             | 9/10000 [00:29<8:29:34,  3.06s/it]tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
{'loss': 2.298, 'grad_norm': 0.0, 'learning_rate': 0, 'epoch': 0.07}                                                                                                  
  0%|                                                                                                                            | 10/10000 [00:32<8:27:08,  3.05s/it]tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
{'loss': 2.4346, 'grad_norm': 0.0, 'learning_rate': 0, 'epoch': 0.08}                                                                                                 
  0%|▏                                                                                                                           | 11/10000 [00:35<8:24:57,  3.03s/it]tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
{'loss': 2.2785, 'grad_norm': 0.0, 'learning_rate': 0, 'epoch': 0.08}                                                                                                 
  0%|▏                                                                                                                           | 12/10000 [00:38<8:22:12,  3.02s/it]tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
{'loss': 2.5054, 'grad_norm': 0.0, 'learning_rate': 0, 'epoch': 0.09}                                                                                                 
  0%|▏                                                                                                                           | 13/10000 [00:41<8:24:34,  3.03s/it]tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
tried to get lr value before scheduler/optimizer started stepping, returning lr=0
{'loss': 2.2641, 'grad_norm': 0.0, 'learning_rate': 0, 'epoch': 0.1}                                                                                                  
{'loss': 2.2935, 'grad_norm': 14.767064094543457, 'learning_rate': 0.0, 'epoch': 0.11}                                                                                
{'loss': 2.3561, 'grad_norm': 14.767064094543457, 'learning_rate': 0.0, 'epoch': 0.12}                                                                                
{'loss': 2.3729, 'grad_norm': 14.767064094543457, 'learning_rate': 0.0, 'epoch': 0.12}                                                                                
{'loss': 2.4506, 'grad_norm': 14.767064094543457, 'learning_rate': 0.0, 'epoch': 0.13}                                                                                
{'loss': 2.3528, 'grad_norm': 14.767064094543457, 'learning_rate': 0.0, 'epoch': 0.14}                                                                                
  0%|▏ 

[rank5]:   File "/home/jiaowei/anaconda3/envs/minicpm/lib/python3.11/site-packages/deepspeed/runtime/fp16/loss_scaler.py", line 175, in update_scale
[rank5]:     raise Exception(
[rank5]: Exception: Current loss scale already at minimum - cannot decrease scale anymore. Exiting run.
  0%|▏                                                                                                                           | 18/10000 [00:59<9:09:37,  3.30s/it]

### 复现方法 | Steps To Reproduce

fintune_lora.sh

### 运行环境 | Environment

```Markdown
- OS: centos 
- Python:3.11.9
- Transformers:4.40.0
- PyTorch:2.1.2
- CUDA (`python -c 'import torch; print(torch.version.cuda)'`): 12.1
```

**Comment by mingo-doer** [comment]
> hi we update the code please try again

我用的应该就是最新的代码 用zero3配置不报错，用zero2 有上述的错误。

**Comment by mingo-doer** [comment]
并且用最新的readme 来加载模型 lora 训练的效果并不好。

**Comment by mingo-doer** [comment]
> hi we update the code please try again

这边方便加个联系方式来沟通一下吗？

**Comment by qyc-98** [comment]
这个应该是训练代码的问题 我建议重新训练一下 再加载试试 我们有微信群 你看看其他issue有提到的

**Comment by mingo-doer** [comment]
> 这个应该是训练代码的问题 我建议重新训练一下 再加载试试 我们有微信群 你看看其他issue有提到的

微信群 二维码过期了，可以更新一下吗？

**Comment by qyc-98** [comment]
![image](https://github.com/user-attachments/assets/a68425aa-330d-4cf1-9ff5-fcc6d0fab81d)

**Comment by mingo-doer** [comment]
> 你的代码里有modules_to_save这些lora初始化时的设定吗

有的，我对比过 是你更新的最新的代码。
我的数据集示例是这样的。我也尝试过不在图像上做红色矩形框标注，直接给坐标，但是因为给坐标直接推理，效果很差，推测在预训练这类任务比较少，所以目前是在图像中用红色矩形框增加明显的信号。
    {
        "id": "2962",
        "image": "Inputbox2.jpg",
        "conversations": [
            {
                "role": "user",
                "content": "<image>\n请先将注意力集中到图片中红色矩形框标记的控件，仅输出该控件的类型。\n\n## 1. 控件类型\n从以下选项中选择并输出控件类型：\n- **勾选框**: 红框内存在可勾选或单选的控件。\n- **下拉选择器**: 右侧有下拉箭头的输入框或下拉菜单。\n- **输入框**: 普通的输入框。\n- **日期选择器**: 可通过点选输入日期的控件，通常有日历图标或日期格式提示。\n- **功能按钮**: 功能按钮，如搜索按钮、提交按钮等。\n- **其他**: 其他类型的控件。\n\n## 3. 输出格式\n仅输出以下类型:\n{\"type\": \"控件类型\"}\n"
            },
            {
                "role": "assistant",
                "content": "{\"type\": \"Inputbox\"}"
            }
        ]
    }

<img width="798" alt="image" src="https://github.com/user-attachments/assets/5bf6ef8a-fe36-4bc5-9d76-e3bc2e709f09">

**Comment by qyc-98** [comment]
建议用bf16训练，比较稳定。然后你这个数据集开头为什么没有<image>这个token呢

**Comment by mingo-doer** [comment]
> 建议用bf16训练，比较稳定。然后你这个数据集开头为什么没有我们示例里面的图像token 这个token呢

好的 谢谢，那我用bf16再试试，其实loss 降的非常快，但是效果并没有看到。
另外有图像token是有的，可能是github markdown 编辑器的原因给吞掉了。
<img width="916" alt="image" src="https://github.com/user-attachments/assets/348d8184-3531-4cb6-85c7-fc11c21525a8">

**Comment by qyc-98** [comment]
嗯嗯 好的好的，我感觉你可以看看参数有没有正常保存，我这里是正常保存的（对比一下保存时的部分参数的value，加载的时候再打印一下）

**Comment by qyc-98** [comment]
针对lora，我刚刚修复了一个问题：我们最后的保存出了问题，你可以试试中间checkpoint的保存结果 ，最后那个保存没有把vpm resampler embed tokens这些权重更新完的模块保存下来，刚刚修改了

**Comment by songge25** [comment]
bf16训练 会提示loss=nan或者inf
